pkgs/build-support/docker/default.nix

```diff
@@ -761,16 +792,27 @@ rec {
   mergeImages = images: runCommand "merge-docker-images"
     {
       inherit images;
-      nativeBuildInputs = [ pigz jq ];
+      nativeBuildInputs = [ file jq ]
+        ++ compressors.none.nativeInputs
+        ++ compressors.gz.nativeInputs
+        ++ compressors.zstd.nativeInputs;
     } ''
     mkdir image inputs
     # Extract images
     repos=()
     manifests=()
+    last_image_mime="application/gzip"
     for item in $images; do
       name=$(basename $item)
       mkdir inputs/$name
-      tar -I pigz -xf $item -C inputs/$name
+
+      last_image_mime=$(file --mime-type -b $item)
+      case $last_image_mime in
+        "application/x-tar") ${compressors.none.decompress};;
+        "application/zstd") ${compressors.zstd.decompress};;
+        "application/gzip") ${compressors.gz.decompress};;
```

Comment:
```suggestion
        "application/gzip") ${compressors.gz.decompress};;
        *) echo "error: unexpected layer type $last_image_mime"; exit 1
          ;;
```

Reply:
Needs `-o pipefail` to fail correctly. Do we have that?

Reply:
Yes. I'm not sure where exactly it is specified, but it appears to be enabled by default in `runCommand`. Adding a `set | grep SHELLOPTS` as the first line prints:

```
merge-docker-images> SHELLOPTS=braceexpand:errexit:hashall:interactive-comments:pipefail
```

While running the tests.